userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -820,9 +823,21 @@ static int32_t load_bpf_file(
 		{
 			if(is_tp_enabled(&(oargs->tp_of_interest), shname))
 			{
-				if(load_tracepoint(handle, shname, data->d_buf, data->d_size) != SCAP_SUCCESS)
+				bool already_attached = false;
+				for (int i = 0; i < handle->m_bpf_prog_cnt && !already_attached; i++)
+				{
+					if (strcmp(handle->m_bpf_progs[i].name, shname) == 0)
+					{
+						already_attached = true;
```

Comment:
Check if the requested tracepoint is already attached (ie: if we already have its name in `m_bpf_progs`).
Avoid attaching it twice.